From c9c758a208cba6e42023136a60a3d46cdb4df629 Mon Sep 17 00:00:00 2001
From: Stefano Rivera <stefano@rivera.za.net>
Date: Mon, 6 Jun 2011 12:04:21 +0200
Subject: [PATCH 2/2] Before download, Package list files are checked with
 prepended $tempdir. At download time, Package list
 files are verified without it. Store both forms in
 file_lists.

---
 debmirror |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/debmirror b/debmirror
index ce641c5..520083e 100755
--- a/debmirror
+++ b/debmirror
@@ -906,9 +906,12 @@ foreach my $dist (keys %distset) {
 		$checksum_type=$1;
 	}
 	elsif (/^ / && defined $checksum_type) {
-      		my ($checksum, $size, $filename) = /^ +([a-z0-9]+) +(\d+) +(.*)$/;
+		my ($checksum, $size, $filename) = /^ +([a-z0-9]+) +(\d+) +(.*)$/;
 		$file_lists{"$tempdir/dists/$dist/$filename"}{$checksum_type} = $checksum;
 		$file_lists{"$tempdir/dists/$dist/$filename"}{size} = $size;
+		# At download time, files are verified without $tempdir
+		$file_lists{"dists/$dist/$filename"}{$checksum_type} = $checksum;
+		$file_lists{"dists/$dist/$filename"}{size} = $size;
 	}
     }
     close RELEASE;
-- 
1.7.4.4

